bin/main

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env ruby
+
+# board that stores the result
+board = ['', '', '', '', '', '', '', '', '']
+
+def welcome()
+  puts ' '
+  puts '|    | |=== |    |===   ===  |\  /| |=== '
+  puts '| /\ | |=== |    |     |   | | \/ | |=== '
+  puts '|/  \| |=== |=== |===   ===  |    | |=== '
+  puts ' '
+  puts ' '
+end
+welcome
+
+# we need to display the bord to the user
+puts 'Welcome to the tic tac toe game'
+def show_board()
+  puts '       X | X | X        '
+  puts '      ***********       '
+  puts '         | O |          '
+  puts '      ***********       '
+  puts '         | O |          '
+end
+
+show_board
+
+puts 'Please press any key to play'
+gets.chomp
+
+# Instantiating the player
+puts 'Player_1 Name: '
+player_one = gets.chomp
+puts "#{player_one} is Player_one with Token \"X\" "
+puts ' '
+puts 'player_2 Name: '
+player_two = gets.chomp
+puts "#{player_two} is Player_two with Token \"O\" "
+
+puts "\n \n"
+
+$players = [
+  { 'name' => [player_one],
+    'token' => 'X' },
+  { 'name' => [player_two],
+    'token' => 'O' }
+]
+
+# puts 'Please choose a position to play!!!'
+def choose_position()
+  puts ' 1 | 2 | 3 '
+  puts '***********'
+  puts ' 4 | 5 | 6 '
+  puts '***********'
+  puts ' 7 | 8 | 9 '
+end
+
+$players = [
+  { 'name' => [player_one],
+    'token' => 'X' },
+  { 'name' => [player_two],
+    'token' => 'O' }
+]
+
+# Players making turns
+def play(board)
+  game_on = true
+  counter = 0
+  player = $players[0]['name']
+
+  while game_on
+
+    if player == $players[0]['name']
+      puts "#{player} please choose you position in the board"
+      choose_position
+      puts '\n'
+      puts 'Position:'
+      gets.chomp
+      player = $players[1]['name']
+    end
+
+    if player == $players[1]['name']
+      puts "#{player} please choose you position in the board"
+      choose_position
+      puts '\n'
+      puts 'Position:'
+      gets.chomp
+      player = $players[1]['name']
+    end
```

Comment:
I think line 88 is preventing the turn switching mechanism.

- **[SUGGESTION]** you can avoid redundant code (i.e asking for player input) in a way that loop look as follows:
```ruby
    while game_on

      puts "#{player} please choose you position in the board"
      choose_position
      puts '\n'
      puts 'Position:'
      gets.chomp
      # here you should Check if the position is valid
      # check if the move give a winner/draw or not
       # if game not done
      player = player.token == 'X' ? $players[1] : $players[0] # To switch turn
      # else
      game_on = false
      # display results 
......
```

- You code either implement that in code, or just insert comments or write empty methods